### PR TITLE
Development to master

### PR DESCRIPTION
**Pull request by SamZand**
PR from dev to master

Main changes:
- Written ```claw.isConnected()``` method to check if the uArm Swift Pro is connected.
- User are able to receive the current state of the robotic claw using the ```claw.getState()``` method.
- Removed inessential methods in Claw class that are not going to be used.
- Improved firmware version and claw state parsing.
- Improved documentation.
https://trello.com/c/SURK0XsN/27-vs4-robot-claw-slice

### CODE REVIEW

File: src/claw.cpp [hunk 1]
```diff
@@ -4,102 +4,123 @@ void Claw::open() {
...
-    /// Move the gripper motor until item is detected
-    hwlib::wait_ms(1000);
-    hwlib::cout << "Item detected!" << hwlib::endlRet;
-    /// Stop the gripper motor
+bool Claw::isConnected() {
+    uartComm << "#n P2203\n";
 
-    /// Set the current position of the motor (if position, depends on if the uArm Swift Pro has an actual actuator)
-    position = 55;
+    /// By giving a null pointer as a method parameter, we save unnecessarily memory space.
```
Comment (elsavira) [inline src/claw.cpp:14]: unnecessary*

File: src/claw.cpp [hunk 2]
```diff
@@ -4,102 +4,123 @@ void Claw::open() {
...
-        /// Move the motors to the specified position
+        /// Check if a timeout occurred. If so, return zero.
+        if ((hwlib::now_us() - lastRead) > readTimeout) {
+            return 0;
+        }
+    }
 
-        /// Set the new position
-        position = destPos;
+    /// Add a \0 character to the response, if the is a response.
```
Comment (elsavira) [inline src/claw.cpp:88]: there*

**Review by elsavira — DISMISSED** [review]
Fix --recursive in this request.

**Review by elsavira — CHANGES_REQUESTED** [review]
Fix Travis test by adding --recursive tag for fetch-build-environment.sh

**Comment by Michelbaartman** [comment]
@Qperquin Changes made by @j00st.

**Review by elsavira — APPROVED** [review]

**Pull request merged**